src/main/java/com/example/music/backend/user/controller/UserController.java

```diff
@@ -5,64 +5,58 @@
 import com.example.music.backend.user.dto.UserDto;
 import com.example.music.backend.user.exception.UserAlreadyExistException;
 import com.example.music.backend.user.service.UserService;
+import lombok.AllArgsConstructor;
 import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.security.crypto.password.PasswordEncoder;
-import org.springframework.stereotype.Controller;
-import org.springframework.ui.Model;
-import org.springframework.validation.Errors;
 import org.springframework.web.bind.annotation.*;
-import org.springframework.web.servlet.ModelAndView;
+
 
 import javax.servlet.http.HttpServletRequest;
 
 import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
 
-@Controller
+@AllArgsConstructor
+@RestController
 public class UserController {
 
     private ApplicationEventPublisher eventPublisher;
     private final UserService service;
     private final PasswordEncoder encoder;
 
-
-    public UserController(UserService service, ApplicationEventPublisher eventPublisher, PasswordEncoder encoder) {
-        this.service = service;
-        this.eventPublisher = eventPublisher;
-        this.encoder = encoder;
-    }
-
-    @PostMapping(produces = APPLICATION_JSON_VALUE, path = "/user/registration")
-    public ModelAndView create(@ModelAttribute("UserDto") UserDto userDto,
-                               HttpServletRequest request, Errors errors) {
-        ModelAndView mav = new ModelAndView("successRegister", "user", userDto);
+    @PostMapping(consumes = APPLICATION_JSON_VALUE, path = "/user/registration")
+    public ResponseEntity<HttpStatus> create(@RequestBody UserDto userDto,
+                                             HttpServletRequest request) {
+        HttpStatus status;
         try {
             String encodedPassword = encoder.encode(userDto.getPassword());
             userDto.setPassword(encodedPassword);
             User saved = service.create(userDto);
             String appUrl = request.getContextPath();
             eventPublisher.publishEvent(new OnRegistrationCompleteEvent(saved, request.getLocale(), appUrl));
+            status = HttpStatus.OK;
         } catch (UserAlreadyExistException uaeEx) {
-            mav.setViewName("registration");
-            mav.addObject("message", "An account for that username/email already exists.");
+            status = HttpStatus.BAD_REQUEST;
```

Comment:
This catch is the same as for RuntimeException, do you really need it?

Reply:
now the service is responsible for this